framework/src/service/ServiceListeners.cpp

```diff
@@ -446,6 +446,15 @@ void ServiceListeners::GetMatchingServiceListeners(const ServiceEvent& evt,
       const LDAPExpr& ldapExpr = sse.GetLDAPExpr();
       if (ldapExpr.IsNull() || ldapExpr.Evaluate(props, false)) {
         set.insert(sse);
+      } else {
+        if (evt.GetType() == ServiceEvent::SERVICE_REGISTERED) {
```

Comment:
should this be an else if instead of an else containing an if statement?

Reply:
modified